src/components/Tooltip/Tooltip.vue

```diff
@@ -0,0 +1,253 @@
+import { useToggle } from '@/use/useToggle';
+import { Portal } from '../Portal';
+import { TooltipOverlay } from './components';
+import type { VueNode } from '@/utilities/types';
+import type { TooltipOverlayProps } from './components/TooltipOverlay/TooltipOverlay.vue';
+import type { Width, Padding, BorderRadius } from './types';
+import styles from '@polaris/components/Tooltip/Tooltip.module.scss';
+
+const HOVER_OUT_TIMEOUT = 150;
+
+export interface TooltipProps {
+  /** Content to display within the tooltip */
+  content?: string;
+  /** Toggle whether the tooltip is visible */
+  active?: boolean | undefined;
+  /** Delay in milliseconds while hovering over an element before the tooltip is visible */
+  hoverDelay?: number;
+  /** Dismiss tooltip when not interacting with its children */
+  dismissOnMouseOut?: TooltipOverlayProps['preventInteraction'];
+  /**
+   * The direction the tooltip tries to display
+   * @default 'above'
+   */
+  preferredPosition?: TooltipOverlayProps['preferredPosition'];
+  /**
+   * The element type to wrap the activator in
+   * @default 'span'
+   */
+  activatorWrapper?: string;
+  /** Visually hidden text for screen readers */
+  accessibilityLabel?: string;
+  /**
+   * Width of content
+   * @default 'default'
+   */
+  width?: Width;
+  /**
+   * Padding of content
+   * @default 'default'
+   */
+  padding?: Padding;
+  /**
+   * Border radius of the tooltip
+   * @default '200'
+   */
+  borderRadius?: BorderRadius;
+  /** Override on the default z-index of 400 */
+  zIndexOverride?: number;
+  /** Whether to render a dotted underline underneath the tooltip's activator */
+  hasUnderline?: boolean;
+  /** Whether the tooltip's content remains open after clicking the activator */
+  persistOnClick?: boolean;
+}
+
+const props = withDefaults(defineProps<TooltipProps>(), {
+  preferredPosition: 'above',
+  activatorWrapper: 'span',
+  width: 'default',
+  padding: 'default',
+  active: undefined,
+});
+
+const emits = defineEmits<{
+  /* Callback fired when the tooltip is activated */
+  open: [];
+  /* Callback fired when the tooltip is dismissed */
+  close: [];
+}>();
+
+const slots = defineSlots<{
+  /** The element to activate the tooltip */
+  default: (_?: VueNode) => any;
+  /** Content to display within the tooltip */
+  content?: (_?: VueNode) => any;
+}>();
+
+const { presenceList, addPresence, removePresence } = useEphemeralPresenceManagerContext();
+const id = useId();
+const {
+  value: active,
+  setTrue: setActiveTrue,
+  setFalse: handleBlur,
+} = useToggle(Boolean(props.active));
+
+const {
+  value: persist,
+  toggle: togglePersisting
+} = useToggle(Boolean(props.active) && Boolean(props.persistOnClick));
+
+const activatorNode = ref<HTMLElement | null>(null);
+const activatorContainer = ref<HTMLElement | null>(null);
+const mouseEntered = ref(false);
+const shouldAnimate = ref(!props.active);
+const hoverDelayTimeout = ref<NodeJS.Timeout | null>(null);
```

Comment:
Is the type is `number` instead?

Reply:
![Screenshot from 2024-01-26 11-35-03](https://github.com/ownego/polaris-vue/assets/56255783/63c71361-27bd-4d08-9d63-b7718d2c6cd0)
Wrong type when assign value :thinking: